### PR TITLE
Android: Update various dependencies

### PR DESCRIPTION
**Pull request by rtiangha**
With Android 15 officially released, figured it was a good time to update some dependencies. There were a couple of minor changes needed due to deprecation, but nothing major.

There is one more change that will be needed to `libadrenotools` to support upcoming Android 15 hardware with 16KiB page sizes (TL;DR: Lime3DS or any app _not_ compiled with `ANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES` enabled won't run on those devices); I just need to submit my PR upstream, which is tricky because the fix needs to be applied to both libadrenotools and one of its submodules so there will be a bit of a propagation delay when it comes to merge timing.

### CODE REVIEW

**Comment by rtiangha** [comment]
`libadrenotools` got the update so hence the new commit as it's also reliant on NDK 27.

**Comment by OpenSauce04** [comment]
FYI, your fork seems to be protected for whatever reason. Please address this for future PRs.

image

File: src/android/app/build.gradle.kts [hunk 1]
```diff
@@ -32,12 +32,12 @@ android {
     ndkVersion = "27.1.12297006"
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_17
-        targetCompatibility = JavaVersion.VERSION_17
+        sourceCompatibility = JavaVersion.VERSION_21
+        targetCompatibility = JavaVersion.VERSION_21
```
Comment (OpenSauce04) [inline src/android/app/build.gradle.kts]: Is there any particular reason why the Java version was updated here?
I tried reverting back to Java 17 while keeping all the other changes in this PR and it builds and runs as I would expect on Android 15.

Comment (rtiangha) [reply src/android/app/build.gradle.kts]: Sorry; that was a copy/paste holdover from my testing fork. JDK 17 should be good until 2029, although who knows when the Android SDK will start to take advantage of features in newer JDKs (FWIW, 21 is LTS too). But that can be dealt with when the time comes. I've reverted it.

**Review by OpenSauce04 — APPROVED** [review]
All looks good to me, builds and runs fine on my end.

I will be manually merging this to avoid either of the commits related to the Java version being pulled in due to them cancelling each-other out and being redundant as a result. Thanks for your contribution.

**Pull request closed without merging**